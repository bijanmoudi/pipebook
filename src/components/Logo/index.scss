/*===================================
=            Logo Styles            =
===================================*/

@import '../../assets/styles/constants/layout';

.logo {
	&,
	&__image {
		display: inline-block;
		vertical-align: middle;
		font-weight: $bold-font-weight;
	}
}
