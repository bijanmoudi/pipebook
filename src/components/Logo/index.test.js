import React from 'react';
import renderer from 'react-test-renderer';
import Logo from './Logo';

const props = {
	id: 'footer-logo',
	width: 50,
	color: '#000000'
};

describe('Testing "Logo" component', () => {
	it('Rendering default logo', () => {
		const tree = renderer.create(<Logo id={props.id} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Rendering width custom color and custom width', () => {
		const tree = renderer
			.create(<Logo id={props.id} width={props.width} color={props.color} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
