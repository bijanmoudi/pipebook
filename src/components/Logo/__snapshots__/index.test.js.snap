// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing "Logo" component Rendering default logo 1`] = `
<span
  className="logo"
>
  <svg
    aria-labelledby="footer-logo-title"
    className="logo__image"
    fill="#f7f7f7"
    height={29}
    id="footer-logo"
    role="img"
    viewBox="0 0 851.9 191.7"
    width={124}
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="footer-logo-title"
    >
      Pipedrive logo
    </title>
    <path
      d="M72.4 38.7c-17 0-26.9 7.7-31.6 12.9-.6-4.6-3.6-10.4-15.3-10.4H0v26.6h10.4c1.7 0 2.3.6 2.3 2.3v121.6H43v-48.8C47.7 147 56.7 153 70.9 153c29.6 0 50.3-23.4 50.3-57.2.1-34.1-19.6-57.1-48.8-57.1m-6.1 87.9c-16.3 0-23.7-15.6-23.7-30.2 0-22.9 12.4-31 24.2-31 14.3 0 24 12.3 24 30.7-.2 21.2-12.5 30.5-24.5 30.5m95.7-4.7V57.4c0-10.9-5.3-16.1-16-16.1h-27.2v26.6h10.4c1.7 0 2.3.6 2.3 2.3v64.2c0 11 5.1 16.1 16 16.1h27.2V124h-10.4c-1.6.2-2.3-.5-2.3-2.1m79.2-83.2c-17 0-26.9 7.7-31.7 12.9-.6-4.6-3.6-10.4-15.3-10.4h-25.5v26.6h10.4c1.7 0 2.3.6 2.3 2.3v121.6H212v-48.8c4.7 4.3 13.9 10.3 27.9 10.3 29.6 0 50.3-23.4 50.3-57.2 0-34.3-19.7-57.3-49-57.3m-6.1 87.9c-16.3 0-23.7-15.6-23.7-30.2 0-22.9 12.4-31 24.2-31 14.3 0 24 12.3 24 30.7-.1 21.2-12.4 30.5-24.5 30.5m112.1-87.9c-32.7 0-55.4 23.6-55.4 57.2 0 33.2 24.6 57.2 58.6 57.2 26.7 0 43-15.9 43.7-16.4l1.3-1.1-13.1-21.9-2 1.9c-.1.1-12.1 11.1-28 11.1-15.1 0-26.3-9.3-29-23.9h71.9l.1-2c0-.3.7-6.7.7-9.9 0-31.2-19.6-52.2-48.8-52.2m-22.8 41.5c3.3-10.7 11.6-16.9 22.9-16.9 9.1 0 16.1 7 17.4 16.9h-40.3zm181.9 41.7V16.3C506.3 5.4 501 .2 490.2.2H463v26.6h10.4c1.7 0 2.3.6 2.3 2.3v19.1c-4.4-4-13.1-9.3-28.3-9.3-29.4 0-50 23.6-50 57.2 0 34.2 19.7 57.2 48.9 57.2 16.7 0 26.4-7.7 31.2-13.1.6 4.7 3.7 10.7 15 10.7h26.2v-26.4h-10.1c-1.7-.3-2.3-1-2.3-2.6m-54.2 4.7c-14.4 0-23.7-12-23.7-30.7 0-18.1 9.7-30.3 24.2-30.3 16.4 0 23.7 15.1 23.7 30.2 0 22.8-12.4 30.8-24.2 30.8m134.2-86.9c-12.6 0-24.3 7.3-31.2 18.9v-2.4c0-9.9-5.4-15-16.1-15h-26v26.6h10.4c1.7 0 2.3.6 2.3 2.3v80.5H556v-43.7c0-5.7.7-11.3 2.1-16.4 4.6-15.1 16.7-20.4 26.4-20.4 3.4 0 6 .4 6 .4l2.6.4V40.6l-1.9-.3c.1-.2-2.4-.6-4.9-.6m50 82.2V57.4c0-10.9-5.3-16.1-16-16.1h-24.6v26.6h7.9c1.7 0 2.3.6 2.3 2.3v64.2c0 11 5.1 16.1 16 16.1h27.2V124h-10.4c-1.6.2-2.4-.5-2.4-2.1m114.5-80.6h-16.1c-9.4 0-14.6 3.3-17.6 11.3l-19.7 55.2c-.7 2-1.3 4.3-1.9 6.6-.6-2.1-1.3-4.6-2-6.6l-19.7-55.2c-3-8.1-7.9-11.3-17.6-11.3h-17.1v26.6h4.4c2.4 0 3.1.9 3.7 2.3l30.6 80.5H712l30.6-80.5c.6-1.4 1.3-2.3 3.7-2.3h4.7V41.3h-.2zm52.3-2.6c-32.7 0-55.4 23.6-55.4 57.2 0 33.2 24.6 57.2 58.6 57.2 26.7 0 43-15.9 43.7-16.4l1.3-1.1-13.1-21.9-2 1.9c-.1.1-12.1 11.1-27.9 11.1-15.1 0-26.3-9.3-29-23.9h71.9l.1-2c0-.3.6-6.7.6-9.9 0-31.2-19.6-52.2-48.8-52.2m-22.8 41.5c3.3-10.7 11.6-16.9 22.9-16.9 9.1 0 16.1 7 17.4 16.9h-40.3zM132.9 0h27v28.4h-27V0zm474.4 0h27v28.4h-27V0z"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Testing "Logo" component Rendering width custom color and custom width 1`] = `
<span
  className="logo"
>
  <svg
    aria-labelledby="footer-logo-title"
    className="logo__image"
    fill="#000000"
    height={11.693548387096774}
    id="footer-logo"
    role="img"
    viewBox="0 0 851.9 191.7"
    width={50}
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="footer-logo-title"
    >
      Pipedrive logo
    </title>
    <path
      d="M72.4 38.7c-17 0-26.9 7.7-31.6 12.9-.6-4.6-3.6-10.4-15.3-10.4H0v26.6h10.4c1.7 0 2.3.6 2.3 2.3v121.6H43v-48.8C47.7 147 56.7 153 70.9 153c29.6 0 50.3-23.4 50.3-57.2.1-34.1-19.6-57.1-48.8-57.1m-6.1 87.9c-16.3 0-23.7-15.6-23.7-30.2 0-22.9 12.4-31 24.2-31 14.3 0 24 12.3 24 30.7-.2 21.2-12.5 30.5-24.5 30.5m95.7-4.7V57.4c0-10.9-5.3-16.1-16-16.1h-27.2v26.6h10.4c1.7 0 2.3.6 2.3 2.3v64.2c0 11 5.1 16.1 16 16.1h27.2V124h-10.4c-1.6.2-2.3-.5-2.3-2.1m79.2-83.2c-17 0-26.9 7.7-31.7 12.9-.6-4.6-3.6-10.4-15.3-10.4h-25.5v26.6h10.4c1.7 0 2.3.6 2.3 2.3v121.6H212v-48.8c4.7 4.3 13.9 10.3 27.9 10.3 29.6 0 50.3-23.4 50.3-57.2 0-34.3-19.7-57.3-49-57.3m-6.1 87.9c-16.3 0-23.7-15.6-23.7-30.2 0-22.9 12.4-31 24.2-31 14.3 0 24 12.3 24 30.7-.1 21.2-12.4 30.5-24.5 30.5m112.1-87.9c-32.7 0-55.4 23.6-55.4 57.2 0 33.2 24.6 57.2 58.6 57.2 26.7 0 43-15.9 43.7-16.4l1.3-1.1-13.1-21.9-2 1.9c-.1.1-12.1 11.1-28 11.1-15.1 0-26.3-9.3-29-23.9h71.9l.1-2c0-.3.7-6.7.7-9.9 0-31.2-19.6-52.2-48.8-52.2m-22.8 41.5c3.3-10.7 11.6-16.9 22.9-16.9 9.1 0 16.1 7 17.4 16.9h-40.3zm181.9 41.7V16.3C506.3 5.4 501 .2 490.2.2H463v26.6h10.4c1.7 0 2.3.6 2.3 2.3v19.1c-4.4-4-13.1-9.3-28.3-9.3-29.4 0-50 23.6-50 57.2 0 34.2 19.7 57.2 48.9 57.2 16.7 0 26.4-7.7 31.2-13.1.6 4.7 3.7 10.7 15 10.7h26.2v-26.4h-10.1c-1.7-.3-2.3-1-2.3-2.6m-54.2 4.7c-14.4 0-23.7-12-23.7-30.7 0-18.1 9.7-30.3 24.2-30.3 16.4 0 23.7 15.1 23.7 30.2 0 22.8-12.4 30.8-24.2 30.8m134.2-86.9c-12.6 0-24.3 7.3-31.2 18.9v-2.4c0-9.9-5.4-15-16.1-15h-26v26.6h10.4c1.7 0 2.3.6 2.3 2.3v80.5H556v-43.7c0-5.7.7-11.3 2.1-16.4 4.6-15.1 16.7-20.4 26.4-20.4 3.4 0 6 .4 6 .4l2.6.4V40.6l-1.9-.3c.1-.2-2.4-.6-4.9-.6m50 82.2V57.4c0-10.9-5.3-16.1-16-16.1h-24.6v26.6h7.9c1.7 0 2.3.6 2.3 2.3v64.2c0 11 5.1 16.1 16 16.1h27.2V124h-10.4c-1.6.2-2.4-.5-2.4-2.1m114.5-80.6h-16.1c-9.4 0-14.6 3.3-17.6 11.3l-19.7 55.2c-.7 2-1.3 4.3-1.9 6.6-.6-2.1-1.3-4.6-2-6.6l-19.7-55.2c-3-8.1-7.9-11.3-17.6-11.3h-17.1v26.6h4.4c2.4 0 3.1.9 3.7 2.3l30.6 80.5H712l30.6-80.5c.6-1.4 1.3-2.3 3.7-2.3h4.7V41.3h-.2zm52.3-2.6c-32.7 0-55.4 23.6-55.4 57.2 0 33.2 24.6 57.2 58.6 57.2 26.7 0 43-15.9 43.7-16.4l1.3-1.1-13.1-21.9-2 1.9c-.1.1-12.1 11.1-27.9 11.1-15.1 0-26.3-9.3-29-23.9h71.9l.1-2c0-.3.6-6.7.6-9.9 0-31.2-19.6-52.2-48.8-52.2m-22.8 41.5c3.3-10.7 11.6-16.9 22.9-16.9 9.1 0 16.1 7 17.4 16.9h-40.3zM132.9 0h27v28.4h-27V0zm474.4 0h27v28.4h-27V0z"
      role="presentation"
    />
  </svg>
</span>
`;
